controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -1,26 +1,43 @@
 package dscinitialization
 
 import (
+	"fmt"
 	"path"
 
 	operatorv1 "github.com/openshift/api/operator/v1"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
 	corev1 "k8s.io/api/core/v1"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
 )
 
 func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
 	switch instance.Spec.ServiceMesh.ManagementState {
 	case operatorv1.Managed:
-		serviceMeshFeatures := feature.ClusterFeaturesHandler(instance, configureServiceMeshFeatures())
 
-		if err := serviceMeshFeatures.Apply(); err != nil {
-			r.Log.Error(err, "failed applying service mesh resources")
-			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
+		capabilities := []*feature.HandlerWithReporter[*dsciv1.DSCInitialization]{
+			r.serviceMeshCapability(instance, serviceMeshCondition(status.ConfiguredReason, "Service Mesh configured")),
+		}
+
+		authzCapability, err := r.authorizationCapability(instance, authorizationCondition(status.ConfiguredReason, "Service Mesh Authorization configured"))
+		if err != nil {
 			return err
 		}
+		capabilities = append(capabilities, authzCapability)
+
+		for _, capability := range capabilities {
+			capabilityErr := capability.Apply()
+			if capabilityErr != nil {
+				r.Log.Error(capabilityErr, "failed applying service mesh resources")
```

Comment:
is here intentionally to use the old message because both 2 capability belong to "service mesh setup", or should it be 
`r.Log.Error(capabilityErr, capabilityErr.Error())`

Reply:
in the case of non-authornio is not installed, `EmptyFeaturesHandler` is added into capabilities, so the Apply() is only doing ReportCondition() part for authorino.?

Reply:
Correct. I thought it's cleaner way of doing things than some `nil` checks and ad-hoc condition settings.

Here's where make somewhat a static reporting, as the handler is NOOP and will not yield any errors. 

```golang
	if !authorinoInstalled {
		authzMissingOperatorCondition := &conditionsv1.Condition{
			Type:    status.CapabilityServiceMeshAuthorization,
			Status:  corev1.ConditionFalse,
			Reason:  status.MissingOperatorReason,
			Message: "Authorino operator is not installed on the cluster, skipping authorization capability",
		}

		return feature.NewHandlerWithReporter(
			// Noop feature handler results in invoking reporter to set MissingOperator condition for authorino
			feature.EmptyFeaturesHandler,
			createCapabilityReporter(r.Client, instance, authzMissingOperatorCondition),
		), nil
	}
```

Reply:
> is here intentionally to use the old message because both 2 capability belong to "service mesh setup", or should it be
r.Log.Error(capabilityErr, capabilityErr.Error())

@zdtsw I missed this part. But yes, for now I would keep it like that - but ideally when capability is introduced as a domain concept we can use structured log/error and carry this info like we do with feature.